dae/dae/utils/statistics.py

```diff
@@ -0,0 +1,61 @@
+from typing import Dict, Any, Tuple
+from collections.abc import MutableMapping
+
+
+class StatsCollection(MutableMapping):
+    """Helper class for collection of variuos statistics.
+
+    This class would be used in the project in places where collection of
+    statistics data about how components of the system work seems appropriate.
+
+    It provides a dict-like interface.
+
+    The keys are tuples of strings. The values could be anything, but usually
+    they are numbers.
+
+    >>> stats = StatsCollection()
+    >>> stats[("a",)] = 1
+    >>> stats[("a",)]
+    1
+    >>> stats.get(("a", 1))
+
+    The keys a treated as a hierarchy. You can get all values whose key's start
+    match the passed key. For example if you add following:
+    >>> stats[("b", "1")] = 42
+    >>> stats[("b", "2")] = 43
+
+    you can get all values whose keys start with ("b",...) using:
+    >>> stats[("b",)]
+    {('b', '1'): 42, ('b', '2'): 43}
```

Comment:
:+1: for the great documentation. Keep it coming!

Reply:
thanks 😄